src/krbd.cc

```diff
@@ -241,12 +241,17 @@ static int wait_for_udev_add(struct udev_monitor *mon, const krbd_spec& spec,
   for (;;) {
     struct pollfd fds[1];
     struct udev_device *dev;
+    int ret = 0;
```

Comment:
okey, I will change it to "int r;" same with others in this krbd.cc.

But BTW, in generic cases, I usually to initialize local variables to 0 or NULL in my code, and use kzalloc rather than kmalloc. @idryomov  and @dillaman  What's your suggestion about it? Is there any pro or con about initializing local variables or not, when we don't have to do.

Reply:
Since you've used kmalloc and kzalloc, I'm assuming you are talking about the kernel.

I don't think you will find a lot of examples of that.  gcc is pretty good about emitting uninitialized variable use warnings and bogus initialization is generally frowned upon.  The kernel has a macro for suppressing false positive warnings without generating extra code: `uninitialized_var(x)` is preferred over initializing the variable just to shut up the compiler.

As far as the additional safety, there is talk about optionally using a gcc plugin that will always initialize all automatic variables: https://lkml.org/lkml/2019/1/23/262.

For kmalloc vs kzalloc, the line is less clear.  Some subsystems actually insist on using kzalloc.  We generally do so too in rbd and ceph code: kzalloc for data structure allocations, kmalloc for  various buffers.

Reply:
@idryomov Thanx for your advice, it helps to me.

Reply:
@yangdongsheng ... and in general, in a C++ environment, the goal would be to always use the tightest scope possible for a variable (which often times is at the point of initial assignment) [1]. 

[1] https://google.github.io/styleguide/cppguide.html#Local_Variables

Reply:
@dillaman Great, that looks clear to me. 👍